src/Microsoft.AspNetCore.SystemWebAdapters.SessionState/RemoteSession/Framework/ReadOnlySessionHandler.cs

```diff
@@ -0,0 +1,35 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Threading.Tasks;
+using System.Web;
+using System.Web.SessionState;
+using Microsoft.AspNetCore.SystemWebAdapters.SessionState.Serialization;
+
+namespace Microsoft.AspNetCore.SystemWebAdapters.SessionState.RemoteSession;
+
+internal sealed class ReadOnlySessionHandler : HttpTaskAsyncHandler, IReadOnlySessionState
+{
+    private readonly ISessionSerializer _serializer;
+
+    public override bool IsReusable => true;
+
+    public ReadOnlySessionHandler(ISessionSerializer serializer)
+    {
+        _serializer = serializer;
+    }
+
+    public override async Task ProcessRequestAsync(HttpContext context)
+    {
+        await ProcessRequestAsync(new HttpContextWrapper(context));
```

Comment:
Same question as before about wrapping the context

Reply:
Same as above - it's to allow mocking for testing

Reply:
Also why the `Application.CompleteRequest()` is not called from the `HttpContextBase` - it's not mockable either.